Source/WebCore/page/Navigation.cpp

```diff
@@ -920,54 +920,58 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
             promiseList.append(WTFMove(promiseAndWrapper.first));
         }
 
-        waitForAllPromises(promiseList, [this, abortController, document, apiMethodTracker, weakThis = WeakPtr { *this }]() mutable {
-            if (!weakThis || abortController->signal().aborted() || !document->isFullyActive() || !m_ongoingNavigateEvent)
-                return;
+        // FIXME: this emulates the behavior of a Promise wrapped around waitForAll, but we may want the real
+        // thing if the ordering-and-transition tests show timing related issues related to this.
+        protectedScriptExecutionContext()->checkedEventLoop()->queueTask(TaskSource::DOMManipulation, [this, promiseList, abortController, document, apiMethodTracker, weakThis = WeakPtr { *this }]() {
+            waitForAllPromises(promiseList, [this, abortController, document, apiMethodTracker, weakThis = WeakPtr { *this }]() mutable {
```

Comment:
Don't capture this.  Add more strongThis-> below

Reply:
Done.